---
title: "RNAseq_Menin_211122"
output: html_document
---

Location of sequencing runs: /pipeline/Runs/NextSeq/211122_NB501056_0722_AHV55MBGXK/ProjectFolders/Project_Christina-Sparbier/Sample_MHC-DLBCL*

# set and create dirs
```{r}
project.dir <- "/dawson_genomics/Projects/Menin/RNAseq_211122_NB501056_0722_AHV55MBGXK"
align.dir <- file.path(project.dir, "alignment")
results.dir <- file.path(project.dir, "results")
logs.dir <- file.path(project.dir, "logs")
plots.dir <- file.path(project.dir, "plots")
sample.dir <- file.path(project.dir, "sample_info")
scripts.dir <- file.path(project.dir, "scripts")
session.dir <- file.path(project.dir, "session_info")

dir.create(align.dir)
dir.create(logs.dir)
dir.create(plots.dir)
dir.create(results.dir)
dir.create(sample.dir)
dir.create(scripts.dir)
dir.create(session.dir)

setwd(project.dir)
```

# submit trim/align/count/fastqc
```{bash}
#run from align dir
cd alignment

for fq in /pipeline/Runs/NextSeq/211122_NB501056_0722_AHV55MBGXK/ProjectFolders/Project_Christina-Sparbier/Sample_MHC-DLBCL*/*fastq.gz
do
sbatch ../scripts/hisat2_hg19dm6.sbatch $fq
done

#do multiqc of fastqcs
module load multiqc/1.8
multiqc fastqc
```

# load into edger
```{r}
library(edgeR)
library(readxl)

# get sample sheet
sample.sheet <- read_excel(file.path(sample.dir, "MGC-CS-3876_211104_DAWSONLAB_DLBCL.xlsx"), sheet = "RNA-seq", n_max = 24)
# set counts files names
count.dir <- file.path(align.dir, "count")
file.names <- paste0(gsub("\\.|\\+", "-", sample.sheet$Sample_name), "_S", seq(2, 25), ".count")
# remove all "+" symbols to avoid syntactical issues
treatment <- gsub("EPZ\\+VTP", "Combo", sample.sheet$Treatment)

# make sample annotation
sample.ann = data.frame(
  "files" = file.names, 
  "samples" = paste0(sample.sheet$Time, "_", treatment, "_", sample.sheet$Replicate, "_S", seq(2, 25)),  
  "treatment" = treatment,
  "cells" = sample.sheet$`Cell type/line`,
  "time" = sample.sheet$Time,
  "group" = paste(sample.sheet$Time, treatment, sep = "_"),
  "replicate" = sample.sheet$Replicate
  )

#write out sample annotation file 
write.table(
  sample.ann,
  file.path(sample.dir, "sample_annotation.txt"),
  sep = "\t",
  quote = FALSE,
  col.names = TRUE
  )

# sequencing and sample information
dge.counts = readDGE(
  file = sample.ann$files, 
  path = count.dir,
  group = sample.ann$group
  )

#change colnames to sample names
colnames(dge.counts$counts) <- sample.ann$samples
```

# write out raw counts
```{r}
write.table(
  dge.counts$counts, 
  file.path(results.dir, "raw_counts_all.txt"),
  sep = "\t",
  quote = F
  )
```

# separate organisms counts objects
```{r}
dros <- which(substr(rownames(dge.counts), 1, 4) == "FBgn")

dm.counts <- dge.counts[dros, ]
hg.counts <- dge.counts[-dros, ]

counts.all <- list(hg.counts, dm.counts)
```


# Filtering out genes with low counts
```{r}
# init variable for cpms list
counts.cpms.list <- list()

# loop through both counts objects to apply same filtering to both
for(counts.i in 1:length(counts.all)){
  colSums(counts.all[[counts.i]]$counts)
  noint = rownames(counts.all[[counts.i]]) %in% c("__no_feature", "__ambiguous", "__too_low_aQual","__not_aligned","__alignment_not_unique")
  table(noint)
  counts.cpms.list[[counts.i]] = cpm(counts.all[[counts.i]])
  keep = rowSums(counts.cpms.list[[counts.i]] > 1 ) >= 3 & !noint
  counts.all[[counts.i]] = counts.all[[counts.i]][keep,]
  dim(counts.all[[counts.i]]$counts)
}

# set names for lists
names(counts.all) <- c("hg_counts", "dm_counts")
names(counts.cpms.list) <- names(counts.all)

# set variable for hg counts table
counts <- counts.all$hg_counts$counts
```

# write out filtered hg counts
```{r}
write.table(
  counts,
  file.path(results.dir, "filtered_counts_hg.txt"),
  sep = "\t",
  quote = FALSE
)
```

# Plot count numbers
```{r}
pdf(file = file.path(plots.dir, "hg_filtered_counts_barplot.pdf"), width = 6, height = 5)
barplot(colSums(counts), las=2, main="Hg19 filtered counts", names.arg = colnames(counts))
dev.off()
```

## add ann info
```{r}
counts.all$hg_counts$samples$treatment = sample.ann$treatment
counts.all$hg_counts$samples$time = sample.ann$time
counts.all$hg_counts$samples$replicate = sample.ann$replicate
```

# MDS plot un-norm counts
```{r}
library(RColorBrewer)
# Setting colours
col.group <- as.factor(counts.all$hg_counts$samples$group)
levels(col.group) <- brewer.pal(nlevels(col.group), "Set1") 
col.group <- as.character(col.group)

pdf(file.path(plots.dir, "MDS_unnorm_counts.pdf"))
MDS = plotMDS(
  counts.all$hg_counts,
  labels = paste0(counts.all$hg_counts$samples$group, "_", counts.all$hg_counts$samples$replicate), 
  col = col.group
  )
dev.off()
```

# calc norm based on Dm
```{r}
d.counts = calcNormFactors(counts.all$dm_counts, method="TMM")
counts.all$hg_counts$samples$norm.factors <- d.counts$samples$norm.factors
hg.norm <- counts.all$hg_counts
```

# MDS plot norm counts
```{r}
pdf(file.path(plots.dir, "MDS_spike-in_norm_counts.pdf"))
MDS = plotMDS(
  hg.norm,
  labels = paste0(hg.norm$samples$group, "_", hg.norm$samples$replicate), 
  col = col.group
  )
dev.off()
```

# make unnorm data to check hg based normalisation
```{r}
hg.unnorm <- counts.all$hg_counts
hg.unnorm$samples$norm.factors <- rep(1, 24)
```

# also check norm based on hg and check MDS
```{r}
hg.counts.norm = calcNormFactors(hg.unnorm, method="TMM")

pdf(file.path(plots.dir, "MDS_hg_norm_counts.pdf"))
MDS = plotMDS(
  hg.counts.norm,
  labels = paste0(hg.norm$samples$group, "_", hg.norm$samples$replicate), 
  col = col.group
  )
dev.off()
```

# do parallel analysis on unnormalised and hg normalised (hg clusterd better than Dm)
```{r}
counts.list <- list("unnorm" = hg.unnorm, "norm" = hg.counts.norm)
```

# set design and contrasts for DE
```{r}
# design table
design = model.matrix(~0+group, data = hg.unnorm$samples)
colnames(design) <- gsub("group", "", colnames(design))
design

contr.matrix <- makeContrasts(
  EPZvsDMSO_d3 = d3_EPZ-d3_DMSO,
  VTPvsDMSO_d3 = d3_VTP-d3_DMSO,
  CombovsDMSO_d3 = d3_Combo-d3_DMSO,
  CombovsEPZ_d3 = d3_Combo-d3_EPZ,
  CombovsVTP_d3 = d3_Combo-d3_VTP,
  EPZvsDMSO_d5 = d5_EPZ-d5_DMSO,
  VTPvsDMSO_d5 = d5_VTP-d5_DMSO,
  CombovsDMSO_d5 = d5_Combo-d5_DMSO,
  CombovsEPZ_d5 = d5_Combo-d5_EPZ,
  CombovsVTP_d5 = d5_Combo-d5_VTP,
  levels = colnames(design)
  )
contr.matrix
```


# EdgeR estimate dispersions 
```{r}
for(norm.i in 1:length(counts.list)){
  counts.list[[norm.i]] = estimateDisp(counts.list[[norm.i]], design)
}
```

# make gene ann table from BM
```{r}
library(biomaRt)
Hsensembl=useMart(
  host='asia.ensembl.org', 
  biomart='ENSEMBL_MART_ENSEMBL',
  dataset='hsapiens_gene_ensembl'
  )

hg.ann = getBM(
  attributes = c("ensembl_gene_id", "hgnc_symbol", "entrezgene_id"), 
  mart = Hsensembl,
  uniqueRows = TRUE
  )
```

# use edgeR for DE
```{r}
ftest.list <- list()
sig.ftest.list <- list()

for(norm.i in 1:length(counts.list)){
  gfit_hg <- glmQLFit(counts.list[[norm.i]], design)
  edger.dir <- file.path(results.dir, names(counts.list)[norm.i], "edger")
  dir.create(edger.dir, recursive = TRUE)
  ftest.list[[norm.i]] <- list()
  
  for(contrast.i in 1:length(colnames(contr.matrix))){
  
    ftest.list[[norm.i]][[contrast.i]] <- glmQLFTest(gfit_hg, contrast = contr.matrix[ , contrast.i])
    ftest.list[[norm.i]][[contrast.i]]$table$ensg <- rownames(ftest.list[[norm.i]][[contrast.i]]$table)
    ftest.list[[norm.i]][[contrast.i]]$table <- merge(ftest.list[[norm.i]][[contrast.i]]$table, hg.ann, by.x = "ensg", by.y = "ensembl_gene_id")
    # write ftest table
    write.table(
      ftest.list[[norm.i]][[contrast.i]]$table,
      file.path(edger.dir, paste0(colnames(contr.matrix)[contrast.i], "_ftest_table.txt")),
      sep = "\t",
      quote = F,
      col.names = T,
      row.names= F
    )
  }
  
  # add contrast names to ftest list
  names(ftest.list[[norm.i]]) <- colnames(contr.matrix)
  
  # make list of significant ftest tables
  sig.ftest.list[[norm.i]] <- list()
  for(contrast.i in 1:length(colnames(contr.matrix))){
   sig.ftest.list[[norm.i]][[contrast.i]] <- subset(ftest.list[[norm.i]][[contrast.i]]$table, PValue < 0.05)
  }

  names(sig.ftest.list[[norm.i]]) <- colnames(contr.matrix)
}
names(ftest.list) <- names(counts.list)
names(sig.ftest.list) <- names(counts.list)
```

# MA plots for each contrast
```{r}
for(norm.i in 1:length(counts.list)){
  ma.dir <- file.path(plots.dir, names(counts.list)[norm.i], "MA")
  dir.create(ma.dir, recursive = TRUE)
  for(contrast.i in 1:length(colnames(contr.matrix))){
    pdf(file.path(ma.dir, paste0(colnames(contr.matrix)[contrast.i], "_MAplot.pdf")))
    maPlot(
      logAbundance = ftest.list[[norm.i]][[contrast.i]]$table$logCPM,
      logFC = ftest.list[[norm.i]][[contrast.i]]$table$logFC,
      main = colnames(contr.matrix)[contrast.i],
      xlab = expression(bold(paste("log"[2], " CPM"))),
      ylab = expression(bold(paste("log"[2], " FC"))),
      col = "black",
      allCol ="red",
      lowess = T
    )
    dev.off()
  }
}
```

# find genes commonly up or down regulated in both treatments
```{r}
library(eulerr)
library(gridExtra)

for(norm.i in 1:length(sig.ftest.list)){
  # get significant DE genes for each single treatment/cell line
  d3.epz <- sig.ftest.list[[norm.i]]$EPZvsDMSO_d3
  d3.vtp <- sig.ftest.list[[norm.i]]$VTPvsDMSO_d3
  d3.combo <- sig.ftest.list[[norm.i]]$CombovsDMSO_d3
  d5.epz <- sig.ftest.list[[norm.i]]$EPZvsDMSO_d5
  d5.vtp <- sig.ftest.list[[norm.i]]$VTPvsDMSO_d5
  d5.combo <- sig.ftest.list[[norm.i]]$CombovsDMSO_d5
  
  # merge cell lines tables on common genes 
  d3.epz.vtp <- merge(d3.epz, d3.vtp, by.x = c("ensg", "hgnc_symbol", "entrezgene_id"), by.y = c("ensg", "hgnc_symbol", "entrezgene_id"))
  d3.epz.combo <- merge(d3.epz, d3.combo, by.x = c("ensg", "hgnc_symbol", "entrezgene_id"), by.y = c("ensg", "hgnc_symbol", "entrezgene_id"))
  d3.vtp.combo <- merge(d3.vtp, d3.combo, by.x = c("ensg", "hgnc_symbol", "entrezgene_id"), by.y = c("ensg", "hgnc_symbol", "entrezgene_id"))
  d5.epz.vtp <- merge(d5.epz, d5.vtp, by.x = c("ensg", "hgnc_symbol", "entrezgene_id"), by.y = c("ensg", "hgnc_symbol", "entrezgene_id"))
  d5.epz.combo <- merge(d5.epz, d5.combo, by.x = c("ensg", "hgnc_symbol", "entrezgene_id"), by.y = c("ensg", "hgnc_symbol", "entrezgene_id"))
  d5.vtp.combo <- merge(d5.vtp, d5.combo, by.x = c("ensg", "hgnc_symbol", "entrezgene_id"), by.y = c("ensg", "hgnc_symbol", "entrezgene_id"))
  
  # rename columns for clarity
  colnames(d3.epz.vtp)[4:11] <- c("EPZ_logFC", "EPZ_logCPM", "EPZ_F", "EPZ_PValue", "VTP_logFC", "VTP_logCPM", "VTP_F", "VTP_PValue")
  colnames(d3.epz.combo)[4:11] <- c("EPZ_logFC", "EPZ_logCPM", "EPZ_F", "EPZ_PValue", "Combo_logFC", "Combo_logCPM", "Combo_F", "Combo_PValue")
  colnames(d3.vtp.combo)[4:11] <- c("VTP_logFC", "VTP_logCPM", "VTP_F", "VTP_PValue", "Combo_logFC", "Combo_logCPM", "Combo_F", "Combo_PValue")
  colnames(d5.epz.vtp)[4:11] <- c("EPZ_logFC", "EPZ_logCPM", "EPZ_F", "EPZ_PValue", "VTP_logFC", "VTP_logCPM", "VTP_F", "VTP_PValue")
  colnames(d5.epz.combo)[4:11] <- c("EPZ_logFC", "EPZ_logCPM", "EPZ_F", "EPZ_PValue", "Combo_logFC", "Combo_logCPM", "Combo_F", "Combo_PValue")
  colnames(d5.vtp.combo)[4:11] <- c("VTP_logFC", "VTP_logCPM", "VTP_F", "VTP_PValue", "Combo_logFC", "Combo_logCPM", "Combo_F", "Combo_PValue")
  
  # make list of merged tables
  merge.list <- list(
    "d3_EPZ_VTP" = d3.epz.vtp,
    "d3_EPZ_Combo" = d3.epz.combo,
    "d3_VTP_Combo" = d3.vtp.combo,
    "d5_EPZ_VTP" = d5.epz.vtp,
    "d5_EPZ_Combo" = d5.epz.combo,
    "d5_VTP_Combo" = d5.vtp.combo
  )
  # write tables in common genes dir
  common.dir <- file.path(results.dir, names(counts.list)[norm.i], "edger/common_genes")
  dir.create(common.dir)
  
  for(merge.i in 1:length(merge.list)){
    # keep only genes with matching directionality of FC
    keep <- which(((merge.list[[merge.i]][4] > 0) + (merge.list[[merge.i]][8] > 0)) != 1)
    merge.list[[merge.i]] <- merge.list[[merge.i]][keep, ]
    
    # write tables in common genes dir
    write.table(
      merge.list[[merge.i]],
      file.path(common.dir, paste0(names(merge.list)[merge.i], "_common_genes_table.txt")),
      sep = "\t",
      quote = F,
      col.names = T,
      row.names = F
    )
  }
  
  # make Eulers of up and down-reg for each cell line
  # separate into up-/down-regulated
  d3.up.genes <- list(
    VTP = unique(d3.vtp$ensg[d3.vtp$logFC > 0]),
    EPZ = unique(d3.epz$ensg[d3.epz$logFC > 0])
  )
  
  d3.down.genes <- list(
    VTP = unique(d3.vtp$ensg[d3.vtp$logFC < 0]),
    EPZ = unique(d3.epz$ensg[d3.epz$logFC < 0])
  )
  
  d5.up.genes <- list(
    VTP = unique(d5.vtp$ensg[d5.vtp$logFC > 0]),
    EPZ = unique(d5.epz$ensg[d5.epz$logFC > 0])
  )
  
  d5.down.genes <- list(
    VTP = unique(d5.vtp$ensg[d5.vtp$logFC < 0]),
    EPZ = unique(d5.epz$ensg[d5.epz$logFC < 0])
  )
  
  pdf(file.path(plots.dir, names(counts.list)[norm.i], "Euler_EPZ_VTP_genes_overlap.pdf"))
  grid.arrange(
    plot(euler(d3.up.genes), quantities = T, fills = c("indianred3", "indianred1"), main = "d3 up-regulated genes"),
    plot(euler(d3.down.genes), quantities = T, fills = c("lightskyblue3", "lightskyblue1"), main = "d3 down-regulated genes"),
    plot(euler(d5.up.genes), quantities = T, fills = c("lightsalmon3", "lightsalmon1"), main = "d5 up-regulated genes"),
    plot(euler(d5.down.genes), quantities = T, fills = c("paleturquoise3", "paleturquoise1"), main = "d5 down-regulated genes")
  )
  dev.off()
}
```

# heatmaps for expression of sig DE genes in each comparison, both LFC ordered and clustered
```{r}
library(viridisLite)
library(gplots)

# make list of normalised logCPMs as matrix for heatmap
log.cpm.list <- list(
  "unnorm" = as.matrix(cpm(counts.list$unnorm, normalized.lib.sizes = TRUE, log = TRUE)),
  "norm" = as.matrix(cpm(counts.list$norm, normalized.lib.sizes = TRUE, log = TRUE))
)

for(norm.i in 1:length(log.cpm.list)){
  
  heatmap.dir <- file.path(plots.dir, names(log.cpm.list)[norm.i], "heatmaps")
  dir.create(heatmap.dir)
  
  for(con.i in 6:length(colnames(contr.matrix))){
    # get samples for contrast, then limit heatmap to contrast samples and sig only for contrast
    contr.levels <- which(t(contr.matrix)[con.i,] != 0)
    contr.samples <- sample.ann$samples[sample.ann$group %in% names(contr.levels)]
    heat.data <- log.cpm.list[[norm.i]][sig.ftest.list[[norm.i]][[con.i]]$ensg, contr.samples]
    # use symbol for rownames, but if unavailable use ensg
    rownames(heat.data) <- sig.ftest.list[[norm.i]][[con.i]]$hgnc_symbol
    no.symbol <- which(sig.ftest.list[[norm.i]][[con.i]]$hgnc_symbol == "")
    rownames(heat.data)[no.symbol] <- sig.ftest.list[[norm.i]][[con.i]]$ensg[no.symbol]
    # remove any duplicates
    heat.unique <- unique(heat.data)
    print(dim(heat.data))
    # heatmap with clustering dendograms
    pdf(file.path(heatmap.dir, paste0(colnames(contr.matrix)[con.i], "_LCPM_clustered_heatmap.pdf")))
    heatmap.2(
      heat.unique,
      cexCol = 0.95,
      cexRow = 0.1,
      margins = c(8, 4),
      main = paste0("logCPM ", colnames(contr.matrix)[con.i]),
      ylab = paste0("Genes (n=", nrow(heat.unique), ")"),
      col = viridis(200, option = "B"),
      key = TRUE,
      breaks = c(seq(min(heat.unique), max(heat.unique), length.out = 201)),
      trace = "none"
    )
    dev.off()
    # heatmap oredered ny LFC in comparison
    heat.data <- heat.data[order(sig.ftest.list[[norm.i]][[con.i]]$logFC, decreasing = TRUE),]
    heat.data <- unique(heat.data)
    pdf(file.path(heatmap.dir, paste0(colnames(contr.matrix)[con.i], "_LFC_ordered_heatmap.pdf")))
    heatmap.2(
      heat.data,
      Rowv = NA,
      cexCol = 0.95,
      cexRow = 0.1,
      margins = c(8, 4),
      main = paste0("logCPM ", colnames(contr.matrix)[con.i]),
      ylab = paste0("Genes (n=", nrow(heat.data), ")"),
      col = viridis(200, option = "B"),
      key = TRUE,
      breaks = c(seq(min(heat.unique), max(heat.unique), length.out = 201)),
      trace = "none"
    )
    dev.off()
  }
}
```

# save session info
```{r}
sesh.info <- capture.output(sessionInfo())
writeLines(
  sesh.info,
  file.path(session.dir, paste0(Sys.Date(), "_RNAseq_211122.txt"))
)
```
